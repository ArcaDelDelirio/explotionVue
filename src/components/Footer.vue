<script>
import store from '../store'
export default {
    name:'Footer',
    props:["users"],
    computed:{
        apiStatus: () => store.state.apiState,
        responseTime: () => store.state.responseTime
    }
}
</script>


<template>
    <footer>
        <div class="flag-api">
            <div :class="{active:apiStatus}" class="status">
                <i><font-awesome-icon icon="circle"/></i>
            </div>
            {{responseTime}}ms
            <i> <font-awesome-icon icon="server"/> API State </i>
        </div>
        <div v-bind:key="user.id" v-for="user in users">
            <i v-if="user.id == 1">User: {{user.name}}</i> 
        </div>
    </footer>
</template>


<style scoped>
    footer{
        display: flex;
        bottom: 0;
    }
    .status{
        color:red;
    }
    .active{
        color:green;
    }
    .flag-api{
        display: flex;
    }
    ul{
       list-style-type: none; 
    }
</style>
