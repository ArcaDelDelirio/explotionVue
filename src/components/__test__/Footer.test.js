import { mount } from '@vue/test-utils';
import Footer from '../Footer';

describe('Footer View', () => {
  it('Render the component', () => {
    const wrapper = mount(Footer);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
