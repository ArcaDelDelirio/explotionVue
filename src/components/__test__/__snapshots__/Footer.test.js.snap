// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer View Render the component 1`] = `
<footer>
  <div class="flag-api">
    <div class="status"><i>
        <font-awesome-icon icon="circle"></font-awesome-icon>
      </i></div>
    0ms
    <i>
      <font-awesome-icon icon="server"></font-awesome-icon> API State
    </i>
  </div>
</footer>
`;
